### PR TITLE
Haskell: Add hlint rule to suggest foldl' over foldl

### PR DESCRIPTION
**Pull request by hurryabit**
`foldl` is lazy in a way that almost never is what you want since it
can cause space leaks without any benefit. `foldl'` does not have this
problem. See https://www.well-typed.com/blog/2014/04/fixing-foldl/ for
more details.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/7897)

### CODE REVIEW

File: .hlint.yaml [hunk 1]
```diff
@@ -53,7 +56,7 @@
 - warn: {lhs: Data.Text.Lazy.readFile, rhs: Data.Text.Extended.readFileUtf8}
 - warn: {lhs: Data.Text.Lazy.writeFile, rhs: Data.Text.Extended.writeFileUtf8}
 - warn: {lhs: System.Environment.setEnv, rhs: System.Environment.Blank.setEnv}
-- warn: {lhs: toNormalizedFilePath, rhs: toNormalizedFilePath'}
+- warn: {lhs: toNormalizedFilePath, rhs: "toNormalizedFilePath'"}
```
Comment (cocreature) [inline .hlint.yaml:59]: Did this not work? I thought I tested this.

**Review by cocreature — APPROVED** [review]
Thanks!

File: .hlint.yaml (hunk 1 above)
Comment (hurryabit) [reply .hlint.yaml:59]: It messes up the yaml syntax highlighting in my editor. My yaml is not the best, but this change doesn't make things worse, does it?

Comment (cocreature) [reply .hlint.yaml:59]: Should be fine, I just got slightly confused.

**Pull request merged**